sovryn-token-bridge/federator/src/main.js

```diff
@@ -71,21 +74,47 @@ const mainFederator = new Federator(
     config,
     log4js.getLogger('MAIN-FEDERATOR'),
     web3,
-    chatBot,
+    p2pNode,
+    chatBot
 );
 
 const sideFederator = new Federator(
     {
         ...config,
         mainchain: config.sidechain,
         sidechain: config.mainchain,
-        storagePath: `${config.storagePath}/side-fed`
+        storagePath: `${config.storagePath}/side-fed`,
     },
     log4js.getLogger('SIDE-FEDERATOR'),
     web3,
-    chatBot,
+    p2pNode,
+    chatBot
 );
 
+p2pNode.net.onMessage(async (msg) => {
+    if (msg.type === MAIN_SIGNATURE_REQUEST) {
+        await handleRequest(msg, mainFederator, MAIN_SIGNATURE_SUBMISSION);
+    } else if (msg.type === SIDE_SIGNATURE_REQUEST) {
+        await handleRequest(msg, sideFederator, SIDE_SIGNATURE_SUBMISSION);
+    }
+});
+
+async function handleRequest(msg, federator, submissionType) {
+    logger.info(
+        `A signature request has been received from ${msg.source.id} for event ${msg.data.log.id}`
+    );
+    try {
+        const { blockNumber, id } = msg.data.log;
+        const signature = await federator.signTransaction({
+            blockNumber,
+            id,
+        });
+        msg.source.send(submissionType, signature);

```

Comment:
As pointed out elsewhere in this review, I think message should also include an identifier for the log (or the whole log) so it can be verified on the other end.